### DISCUSSION TITLE
run curobo example realsense_reacher.py with real panda robot

### DISCUSSION THREAD

**Discussion by from1tomax** [NVlabs/curobo#414] [Research Q&A] [answered]
Hello, I am quite amazed at the real-time obticle avoiding ability using curobo in the example demo **realsense_reacher.py** .
However I want to go on to run it with real panda robot , as it is only written in python script, I would like to know how to run the demo with real robot, is it necessary to wrapper it into ROS stucture? Thanks in advance

**Comment by balakumar-s** [comment, marked as answer]
We have an example workflow that does this using ROS2 (should work on Franka Panda as well): https://nvidia-isaac-ros.github.io/reference_workflows/isaac_manipulator/tutorials/tutorial_e2e.html

This is also possible without ROS, but you would have to integrate camera stream, connect it to nvblox and then share the sdf data to cuRobo. And any motions generated by cuRobo will need to be sent to the robot.
